tools/build/libwebrtc/build.ps1

```diff
@@ -0,0 +1,282 @@
+
+# Get the root folder of MixedReality-WebRTC
+function Get-RootFolder
+{
+    $repoRoot = Join-Path $PSScriptRoot "../../../" -Resolve
+    return $repoRoot
+}
+
+# Get the <root>/external/ folder
+function Get-ExternalFolder
+{
+    $repoRoot = Get-RootFolder
+    $externalFolder = Join-Path $repoRoot "external/"
+    return $externalFolder
+}
+
+# Write a message for a task starting, with a specific color to make it
+# stand out between long-running tasks themselves outputting lots of info.
+function Write-TaskStart([string]$message)
+{
+    Write-Host -Foreground Magenta "`n$message"
+}
+
+# Write a failure message without generating any Powershell error.
+function Write-Failure([string]$message)
+{
+    Write-Host -NoNewLine -Foreground Red "Error: "
+    Write-Host $message
+}
+
+# Write a key-value pair with higlight on the value.
+function Write-KeyValue([string]$key, [string]$value)
+{
+    Write-Host -NoNewLine $key
+    Write-Host -Foreground Blue $value
+}
+
+# Initialize the environment variables
+function Initialize-BuildEnvironment
+{
+    Write-TaskStart "Setting up environment..."
+
+    $repoRoot = Get-RootFolder
+    Write-KeyValue "Repository root: " $repoRoot
+
+    # Add depot_tools folder to PATH
+    $externalFolder = Join-Path $repoRoot "external/"
+    $depotToolsPath = Join-Path $externalFolder "depot_tools"
+    if (!($env:PATH.Contains($depotToolsPath)))
+    {
+        # Be sure to append, not prepend. depot_tools/ contains some git version,
+        # and prepending will make that version the default one, which may confuse
+        # some users as e.g. on Windows it won't have the credential manager set up
+        # like the one the user had installed already.
+        $env:PATH = "$env:PATH;$depotToolsPath"
```

Comment:
This seems to change the environment of the calling process. For cleanliness, does PS have a way to keep the changes in the scope of the script only? Or to fork a subshell with the changed env?

Reply:
Doesn't look like there is a good solution out of the box like on Unix : https://stackoverflow.com/questions/1420719/powershell-setting-an-environment-variable-for-a-single-command-only

I think the way to go is to run jobs :
- https://www.reddit.com/r/PowerShell/comments/qvk0d/powershell_fork/
- https://powerscripting.wordpress.com/2010/06/03/a-look-at-powershell-jobs-part-1/

Reply:
Fixed with scope.